ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLFlowAgentRunner.java

```diff
@@ -260,6 +265,7 @@ Tool createTool(MLToolSpec toolSpec) {
         if (toolSpec.getName() != null) {
             tool.setName(toolSpec.getName());
         }
+        tool.setTenantId(toolSpec.getTenantId());
```

Comment:
Why need to pass tenant id to Tool?

Reply:
Adding new method to pass parameters between agent and tools , tools and tools is not extensible/scalable. If we are going to add 100s new parameters, should we add 100s new set/get methods ?

I personally prefer to pass tenant id in `Tool.run(...)` method. For now we only pass Map of parameters which is mutable. For such immutable parameters, we can add a new Immutable map like 
```
default <T> void run(Map<String, String> parameters, ImmutableMap<String, String> configurations, ActionListener<T> listener) {};
```

Reply:
If there is a requirement for multiple configuration to be added to a tool, we could consider creating a class for configuration and set that to abstract class. 

However, given that tenant ID is a runtime parameter which could change based on API invocation it makes sense to add it to `run` method instead of adding to abstract class.

Reply:
A more scalable method of passing parameters like this would be a key-value map that we could add to.

That said,

> tenant ID is a runtime parameter which could change based on API invocation

I'm a bit confused here. Isn't tenant ID always the same for a given customer? We're using it as a key in our metadata store.  When does it change?